File: e101/python/solution.py
class Solution:
    """
    @param A: a list of integers
    @return an integer
    """
    def removeDuplicates(self, A):
        # write your code here
        if A is None or len(A) == 0:
            return 0
        value = A[0]
        count = 1
        newIdx = 0
        for i in range(1, len(A)):
            if A[i] != value:
                newIdx += 1
                A[newIdx] = A[i]
                value = A[i]
                count = 1
            else:
                count += 1
                if count <= 2:
                    newIdx += 1
                    A[newIdx] = A[i]
                    value = A[i]
                else:
                    pass
        return newIdx + 1

File: e101/java/Solution.java
public class Solution {
    /**
     * @param A: a array of integers
     * @return : return an integer
     */
    public int removeDuplicates(int[] nums) {
        // write your code here
        if (nums == null || nums.length == 0)
            return 0;
        int value = nums[0];
        int count = 1;
        int j = 0;
        
        for (int i=1; i<nums.length; i++) {
            if (nums[i] != value) {
                j++;
                nums[j] = nums[i];
                value = nums[i];
                count = 1;
            }
            else { //nums[i] == value
                count++;
                if (count <= 2) {
                    j++;
                    nums[j] = nums[i];
                    value = nums[i];
                }
                else { //count > 2
                    continue;
                }
            }
        }

        return j+1;
    }
}
